UPGRADING.md

```diff
@@ -0,0 +1,121 @@
+
+## V3
+
+This release brings `level-js` up to par with latest [`levelup`] (v2), [`abstract-leveldown`] (v5) and IndexedDB Second Edition. It targets modern `browserify`, preferring [`Buffer`] (which is the primary binary type in the Level ecosystem) over `ArrayBuffer`. Lastly, [`IDBWrapper`] has been replaced with straight IndexedDB code.
+
+### Usage with [`levelup`]
+
+Usage has changed to:
+
+```js
+const levelup = require('levelup')
+const leveljs = require('leveljs')
+
+const db = levelup(leveljs('mydb'))
+```
+
+From the old:
+
+```js
+const db = levelup('mydb', { db: leveljs })
+```
+
+Friendly reminder: encodings have moved from `levelup` to `encoding-down`. To get identical functionality to `levelup` < 2, please use the `level-browserify` convenience package or wrap `level-js` with `encoding-down`:
+
+```js
+const encoding = require('encoding-down')
+const db = levelup(encoding(leveljs('mydb')))
+```
+
+### New database prefix
+
+The default prefix of the [`IDBDatabase`](https://developer.mozilla.org/en-US/docs/Web/API/IDBDatabase) name has changed from `IDBWrapper-` to `level-js-`. To access databases created using `level-js` < 3, pass a custom prefix to the `level-js` constructor:
+
+```js
+const db = levelup(leveljs('mydb', { prefix: 'IDBWrapper-' }))
+```
+
+### Browser support
+
+As a result of removing [`IDBWrapper`], only modern browsers with a non-prefixed `window.indexedDB` are supported in this release. The current test matrix of `level-js` includes the latest versions of Chrome, Firefox, Safari, Edge and IE.
+
+:fire: Internet Explorer 10 is no longer supported.
+
+### Support all value types of the [structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm)
+
+This means you can store almost any JS type without the need for [`encoding-down`]. Depending on the environment, this includes:
+
+-   Number, including `NaN`, `Infinity` and `-Infinity`
+-   String, Boolean, Date, RegExp, Array, Object
+-   ArrayBuffer or a view thereof (typed arrays);
+-   Map, Set, Blob, File, FileList, ImageData (limited support).
+
+In addition `level-js` now stores `Buffer` values without transformation. This works in all target environments because modern `Buffer` is a subclass of `Uint8Array`, meaning such values can be passed to `IndexedDB` as-is.
+
+If the environment does not support a type, it will throw an error which `level-js` catches and passes to the callbacks of `put` or `batch`. For example, IE does not support typed array values. At the time of writing, Chrome is the only browser that supports all types listed above.
+
+Due to the special meaning that `null` and `undefined` have in `abstract-leveldown` iterators and Node.js streams, values of this type are converted to empty strings prior to storage.
+
+:point_up: The structured clone algorithm can be slower than `JSON.stringify`, so don't to quick to eschew `encoding-down`.
```

Comment:
to => too

Reply:
When wrapping with level-browserify, maybe we should provide an option to use encoding-down or not?

Reply:
IMO `levelup(leveljs())` is just fine, or even better for composition than options.